src/aws.rs

```diff
@@ -371,6 +385,39 @@ impl ObjectStore for AmazonS3 {
             )
             .await?)
     }
+
+    async fn copy(&self, from: &Path, to: &Path) -> Result<()> {
+        let from = from.as_ref();
+        let to = to.as_ref();
+        let bucket_name = self.bucket_name.clone();
+
+        let request_factory = move || rusoto_s3::CopyObjectRequest {
+            bucket: bucket_name.clone(),
+            copy_source: format!("{}/{}", &bucket_name, from),
+            key: to.to_string(),
+            ..Default::default()
+        };
+
+        let s3 = self.client().await;
+
+        s3_request(move || {
+            let (s3, request_factory) = (s3.clone(), request_factory.clone());
+
+            async move { s3.copy_object(request_factory()).await }
+        })
+        .await
+        .context(UnableToCopyObjectSnafu {
+            bucket: &self.bucket_name,
+            from,
+            to,
+        })?;
+
+        Ok(())
+    }
+
+    async fn rename_no_replace(&self, _source: &Path, _dest: &Path) -> Result<()> {
```

Comment:
As mentioned below, I wonder if rather than a rename_no_replace, which no object stores actually support, if `copy_if_not_exists` would be sufficient?

Reply:
Yeah that makes sense. `copy_if_not_exist` is sufficient for delta-rs.